src/app/services/networkManagerService/networkManager.spec.js

```diff
@@ -15,37 +15,68 @@ describe('network Manager service', () => {
 
   beforeEach(angular.mock.module('ngCordova.plugins.network'));
 
-  beforeEach(inject((_$cordovaNetwork_, _networkManagerService_) => {
+  beforeEach(inject((_$cordovaNetwork_, _networkManagerService_, _$window_) => {
     $cordovaNetwork = _$cordovaNetwork_;
     networkManagerService = _networkManagerService_;
-
-    window.Connection = {
+    $window = _$window_;
+    navigator = $window.navigator;
+    $window.Connection = {
       UNKNOWN: 'UNKNOWN',
       NONE: 'NONE',
       CELL_2G: 'CELL_2G'
     };
-
-    navigator.connection = {
-      type: window.Connection.NONE
-    };
+    navigator.connection = { type: $window.Connection.NONE};
   }));
 
-  it('should return isOnline', () => {
-    navigator.connection.type = window.Connection.CELL_2G;
-    expect($cordovaNetwork.isOnline()).to.be.true;
+  it('should be online', () => {
+    ionic.Platform.isWebView = () => false;
+
+    let isOnline = networkManagerService.isOnline();
+
+    expect(isOnline).to.be.true;
   });
 
-  it('should return isOffline', () => {
-    navigator.connection.type = window.Connection.UNKNOWN;
-    expect($cordovaNetwork.isOnline()).to.be.false;
+  it('should be online when is a webview and has connection', () => {
+    ionic.Platform.isWebView = () => true;
+    navigator.connection = { type: $window.Connection.CELL_2G};
+
+    let isOnline = networkManagerService.isOnline();
+
+    expect(isOnline).to.be.true;
   });
 
-  it('should return the network type', () => {
-    expect($cordovaNetwork.getNetwork()).to.be.equal(window.Connection.NONE);
+  it('should not be online when is a webview and has no connection', () => {
```

Comment:
Qué te parece si quitamos el "is a" y metemos "when webview and has no connection"? Aplcaría a resto de tests

Reply:
ok